### PR TITLE
Working with multiple files

### LINKED ISSUE #175: Feature Request: Allow Sequences of Files
I think that this would only make sense to implement if #168 is implemented, and it may be easiest to implement at the same time as that one.

I sometimes receive sequences of files that are related.  Examples include: scientific journal articles and supplements to the journal article; sets of images from many sources (examples are data images taken in sequence, images taken of presentations at scientific conferences, and investor presentations).

As WPD supports multiple axes-- and if #168 is implemented, multiple pages, it would be helpful to also support multiple input files.  The rationale is that extracting data from a set of related files at the same time improves the consistency.  I can ensure that the same names are used for everything more easily if I'm jumping around within a single session than load/check/unload/load/modify.

If implemented, I would imagine that instead of just a page selector as in #168, there would be a file and page selector (pull down for file; pull down for page within file if the file is a multi-page type and has more than one page).  Other interfaces could work, too, but that is the one that first occurs to me.

### PR DESCRIPTION
**Pull request by ewchow**
Implemented support for working with multiple files (resolves  #175).

Summary of changes:
- Allow multiple files to be selected at file upload
- Created file selection dropdown
    - Currently under Image tree item
- Axes, datasets, and measurements respect file and page
- Current page is retained between switching files (if applicable)
- Save/Load project supports multiple files
    - Expanded project saving and loading logic to read and write file and page metadata
- Moved PageManager.js from javascript/core to javascript/controllers
- Refactored various PageManager function logic

### CODE REVIEW

File: app/javascript/controllers/fileManager.js [hunk 1]
```diff
@@ -0,0 +1,410 @@
...
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Affero General Public License for more details.
+
+    You should have received a copy of the GNU Affero General Public License
+    along with WebPlotDigitizer.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+var wpd = wpd || {};
+
+wpd.utils = (function() {
```
Comment (ankitrohatgi) [inline app/javascript/controllers/fileManager.js]: Is this only to be used in the `FileManager`? Should this live in a separate file?

File: app/javascript/controllers/pageManager.js [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+        this.$pageNavInput.setAttribute('max', this.maxPage);
+
+        return this;
+    }
+
+    get() {
+        return this.handle;
+    }
+
+    getPage() {}
```
Comment (ankitrohatgi) [inline app/javascript/controllers/pageManager.js]: Was this intentionally left empty?

**Comment by ankitrohatgi** [comment]
Just had a minor nit-picky comment. Looks good otherwise.

It's hard to catch corner cases in a large PR like this so I am going to assume that you tested out all this fairly well.

I should perhaps consider improving test coverage at some point :(

File: app/javascript/controllers/fileManager.js (hunk 1 above)
Comment (ewchow) [reply app/javascript/controllers/fileManager.js]: Ah good catch, it's actually used in `fileManager.js` and `pageManager.js` - I'll move it out to its own file

Comment (ewchow) [reply app/javascript/controllers/fileManager.js]: All set

**Comment by billdenney** [comment]
Thanks @ewchow and @ankitrohatgi!

**Comment by ewchow** [comment]
Yeah, more test coverage would make it less onerous to QA. I've been vigilant with manual testing, but alas, human after all :/

Just to make sure, I did another run through the workflows, and I did find a bug. I'll be fixing that in a new PR shortly. Thanks!
